src/ssl.rs

```diff
@@ -0,0 +1,52 @@
+use crate::error::Result;
+use std::io::Read;
+use std::path::Path;
+
+#[derive(Clone)]
+pub struct Certificate(pub (crate) reqwest::Certificate);
+
+#[derive(Clone)]
+pub struct Ssl {
+    pub root_cert: Option<Certificate>,
+}
+
+impl Default for Ssl {
+    fn default() -> Self {
+        Ssl {
+            root_cert: None,
+        }
+    }
+}
+
+impl Ssl {
+    pub fn read_pem<P: AsRef<Path> + std::fmt::Display>(root_certificate_path: &P) -> Result<Certificate> {
```

Comment:
`Display` can be removed as `Path` has [display](https://doc.rust-lang.org/std/path/struct.Path.html#method.display).

Reply:
Done.